### PR TITLE
Fix issue with McMaster.DotNetExe returning non-existing path to dotnet on Linux Alpine environment

### PR DESCRIPTION
**Pull request by alexeypukhov**
## Why

On Linux Alpine docker image (alpine.dockerfile) DotNetExe.FullPathOrDefault() returns non-existing path to dotnet: "/usr/local/share/dotnet/dotnet"

Discovered while working and required for https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/4783 - [proof link to the job](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/actions/runs/21189425603/job/60952289076)

## What

If the path returned by the tool is not expected, fallback to just `"dotnet"` giving it a chance for the environment to pickup

## Tests

Will be tested by current build ci and the success of https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/4783 build

## Checklist

- ~[ ] `CHANGELOG.md` is updated.~
- ~[ ] Documentation is updated.~
- ~[ ] New features are covered by tests.~

### CODE REVIEW

File: tools/DependencyListGenerator/DotNetOutdated/Services/DotNetRunner.cs [hunk 1]
```diff
@@ -11,7 +12,18 @@ public static class DotNetRunner
...
     public static RunStatus Run(string workingDirectory, string[] arguments)
     {
-        var psi = new ProcessStartInfo(DotNetExe.FullPathOrDefault(), string.Join(" ", arguments))
+        // 1. Get the path from the library
+        var dotnetPath = DotNetExe.FullPathOrDefault();
+
+        // 2. Check if the file actually exists. If not, just use "dotnet"
+        // and let the OS find it in the PATH.
+        // on Linux DotNetExe.FullPathOrDefault() may wrongly return "/usr/local/share/dotnet/dotnet"
+        if (dotnetPath == null || !File.Exists(dotnetPath))
```
Comment (iskiselev) [inline tools/DependencyListGenerator/DotNetOutdated/Services/DotNetRunner.cs]: `DotNetExe.FullPathOrDefault()` declared as non-null, so we may not need to test it with null.

**Comment by iskiselev** [comment]
Created issues: natemcmaster/CommandLineUtils#600

File: tools/DependencyListGenerator/DotNetOutdated/Services/DotNetRunner.cs (hunk 1 above)
Comment (alexeypukhov) [reply tools/DependencyListGenerator/DotNetOutdated/Services/DotNetRunner.cs]: Right! Fixed

File: tools/DependencyListGenerator/DotNetOutdated/Services/DotNetRunner.cs [hunk 2]
```diff
@@ -11,7 +12,18 @@ public static class DotNetRunner
 {
     public static RunStatus Run(string workingDirectory, string[] arguments)
     {
-        var psi = new ProcessStartInfo(DotNetExe.FullPathOrDefault(), string.Join(" ", arguments))
+        // 1. Get the path from the library
```
Comment (Kielek) [inline tools/DependencyListGenerator/DotNetOutdated/Services/DotNetRunner.cs:15]: Can you put here link to the issue create by @iskiselev.
When updated we can remove local fix.

Comment (alexeypukhov) [reply tools/DependencyListGenerator/DotNetOutdated/Services/DotNetRunner.cs:15]: Done!
